src/Groups/GAPGroups.jl

```diff
@@ -1424,9 +1452,32 @@ function set_prime_of_pgroup(G::GAPGroup, p::IntegerUnion)
   set__prime_of_pgroup(G, GAP.Obj(p))
 end
 
+"""
+    rank(G::GAPGroup)
+
+Return the rank of the group `G`, i.e., the minimal size of a generating set.
```

Comment:
`GrpAbFinGen` defines `rank` as the rank of the free part. Thus torsion groups created with `abelian_group` have always rank zero. If we really want to define `rank` for `GAPGroup` as the cardinality of a minimal generating set then perhaps we should mention this.